src/components/UserSelector.tsx

```diff
@@ -2,39 +1,97 @@
+import { useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import classNames from 'classnames';
+
+type Props = {
+  users: User[];
+  setUserSelected: (user: User) => void;
+  userSelected: User | null;
+  handleUserPosts: (user: User) => void;
+  setShowSideBar: (item: number | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  setUserSelected,
+  userSelected,
+  handleUserPosts,
+  setShowSideBar,
+}) => {
+  const [showUsers, setShowUsers] = useState(false);
+  const dropdownMenuRef = useRef<HTMLDivElement | null>(null);
+  const dropdownTriggerRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownMenuRef.current &&
+        !dropdownMenuRef.current.contains(event.target as Node) &&
+        dropdownTriggerRef.current &&
+        !dropdownTriggerRef.current.contains(event.target as Node)
+      ) {
+        setShowUsers(false);
+      }
+    };
+
+    document.addEventListener('click', handleClickOutside);
```

Comment:
You are interacting with the DOM directly by adding an event listener to the document. This is not recommended in React. Instead, you should use React's synthetic events system. This is a code quality issue.